dask/dataframe/io.py

```diff
@@ -241,12 +243,28 @@ def from_array(x, chunksize=50000):
     name = next(from_array_names)
     dsk = dict(((name, i), (pd.DataFrame,
                              (getitem, x,
-                             (slice(i * chunksize, (i + 1) * chunksize),))))
-            for i in range(0, int(ceil(float(len(x)) / chunksize))))
+                              slice(i * chunksize, (i + 1) * chunksize))))
+            for i in range(0, int(ceil(len(x) / chunksize))))
 
     return DataFrame(dsk, name, columns, divisions)
 
 
+from_dataframe_names = ('from-dataframe-%d' % i for i in count(1))
+
+
+def from_dataframe(df, npartitions=10):
+    nrows = len(df)
+    chunksize = int(ceil(nrows / npartitions))
+    divisions = tuple(range(chunksize, nrows, chunksize))
+    name = next(from_dataframe_names)
+    iloc = df.iloc
+    dsk = dict(((name, i),
+                (getitem, iloc, slice(i * chunksize, (i + 1) * chunksize)))
+               for i in range(npartitions))
+    return DataFrame(dsk, name, tuple(df.columns), divisions)
```

Comment:
Ideally `divisions` matches values in the underlying index, so if `df.index` is a datetime index then the values in divisions would be `pd.Timestamp` objects.  To achieve this I think that we'll have to sort the underlying dataframe ahead of time. 

I would also just insert the dataframe directly into the graph, rather than trigger computation to insert it.  The `iloc` calls should be fast enough to happen at graph creation time and it seems more explicit to include the data directly.